initialization/initialization.go

```diff
@@ -312,6 +309,11 @@ func (init *Initializer) Initialize(ctx context.Context) error {
 	}
 
 	if init.nonce.Load() != nil {
+		init.logger.Info("initialization: completed, found nonce", zap.Uint64("nonce", *init.nonce.Load()))
+		return nil
+	}
+	if layout.FirstFileIdx > 0 || layout.LastFileIdx < init.opts.TotalFiles(init.cfg.LabelsPerUnit)-1 {
+		init.logger.Info("initialization: no nonce found while computing labels")
 		return nil
 	}
```

Comment:
How about adding a method to layout struct:
```go
func (l filesLayout) NumFiles() int {
    return l.LasFileIdx - l.FirstFileIdx + 1
}
```

the code would become more readable I think:
```suggestion
	if layout.NumFiles() < init.opts.TotalFiles(init.cfg.LabelsPerUnit) {
		init.logger.Info("initialization: no nonce found while computing labels")
		return nil
	}
```